ecl/hthor/hthorkey.cpp

```diff
@@ -2902,10 +2903,14 @@ class CJoinGroup : implements IInterface, public CInterface
     inline void noteEnd()
     {
         assertex(!complete());
+        //Another completing group could cause this group to be processed once endMarkersPending is set to 0
+        //So link this object to ensure it is not disposed of while this function is executing
+        Link();
```

Comment:
I think onComplete may throw exceptions - for example on limits being hit - so better to use a Linked<> variable

Reply:
Agreed